spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/ZipkinConfigurations.java

```diff
@@ -157,10 +147,12 @@ static class OpenTelemetryConfiguration {
 
 		@Bean
 		@ConditionalOnMissingBean
-		@ConditionalOnBean(Sender.class)
+		@ConditionalOnBean(BytesMessageSender.class)
 		@ConditionalOnEnabledTracing
-		ZipkinSpanExporter zipkinSpanExporter(BytesEncoder<Span> encoder, Sender sender) {
-			return ZipkinSpanExporter.builder().setEncoder(encoder).setSender(sender).build();
+		ZipkinSpanExporter zipkinSpanExporter(BytesMessageSender sender, Optional<BytesEncoder<Span>> encoder) {
```

Comment:
we *could* swap out this `Optional<BytesEncoder<Span>> encoder` for a bean contained inside OpenTelemetryConfig, like what used to be top-level. When you have a dep on otel's zipkin exporter, you already have the classes needed for below. However, brave will not bring in these classes, and it wouldn't be great to force normal users to do that when they don't need them.

```java
@Bean
	@ConditionalOnMissingBean
	public BytesEncoder<Span> spanBytesEncoder() {
		return SpanBytesEncoder.JSON_V2;
	}
```